### PR TITLE
[ENH] Update subworkflow preproc_t1

### PR DESCRIPTION
**Pull request by anroy1**
## Type of improvement

- [ ] Documentation
- [ ] Development tools (e.g. linter, formatter, etc.)
- [ ] Development container
- [ ] Global update (please specify)
- [x] Other (please specify)

## Describe your improvement

- Adding .map() function to input channel structure to correctly manage empty optional channels
- Removing the ants BET checks (template and probability map)

## Checklist before requesting a review

- [x] Ensure the syntax is correct (**EditorConfig** and **Prettier** must pass)
- [x] Run the test suites if your changes affect any module
- [x] Regenerate the **Poetry** lock file if you have updated the dependencies
- [x] Ensure the documentation is up-to-date

### CODE REVIEW

File: subworkflows/nf-neuro/preproc_t1/main.nf [hunk 1]
```diff
@@ -51,9 +65,10 @@ workflow PREPROC_T1 {
         }
 
         else {
-            ch_template = ch_template.ifEmpty(Channel.error('Template is required for ANTS registration'))
-            ch_probability_map = ch_probability_map.ifEmpty(Channel.error('Probability map is required for ANTS registration'))
-            ch_bet = IMAGE_RESAMPLE.out.image.join(ch_template).join(ch_probability_map)
+            ch_bet = IMAGE_RESAMPLE.out.image
+                .join(ch_template)
+                .join(ch_probability_map)
```
Comment (gagnonanthony) [inline subworkflows/nf-neuro/preproc_t1/main.nf:70]: Why did you remove the `remainder: true` ?

**Comment by AlexVCaron** [comment]
Closes AlexVCaron/nf-neuro#11

File: subworkflows/nf-neuro/preproc_t1/main.nf (hunk 1 above)
Comment (AlexVCaron) [reply subworkflows/nf-neuro/preproc_t1/main.nf:70]: Don't know if this was answered. I assume yes, but will give a reason for posterity : the template and probabilities aren't optional for BETCROP_ANTSBET, so we should not have to `remainder` here.

I would agree to add the error checking. However, I also thinks it is not the job of the subworkflow to do input validation. It's the job of the main pipeline.

Comment (gagnonanthony) [reply subworkflows/nf-neuro/preproc_t1/main.nf:70]: Agreed! I was more curious about the sudden removal of the `remainder`, but that makes perfect sense!

Comment (anroy1) [reply subworkflows/nf-neuro/preproc_t1/main.nf:70]: I remember we had the conversation of removing the ifempty checks. Would you like me to add them back?

Comment (AlexVCaron) [reply subworkflows/nf-neuro/preproc_t1/main.nf:70]: No, keep that like that for now

**Review by AlexVCaron — APPROVED** [review]
All good on my end @arnaudbore should we have a discussion on optional input definition on beginning next week ?

**Review by arnaudbore — APPROVED** [review]
GTG

**Comment by AlexVCaron** [comment]
I will add the comments on the joins with remainders in a separate PR
